cmd/thanos/compact.go

```diff
@@ -458,6 +459,51 @@ func runCompact(
 		return cleanPartialMarked()
 	}
 
+	if conf.compactionProgressMetrics {
+		g.Add(func() error {
+			ps := compact.NewCompactionProgressCalculator(reg, tsdbPlanner)
+			var ds *compact.DownsampleProgressCalculator
+			if !conf.disableDownsampling {
+				ds = compact.NewDownsampleProgressCalculator(reg)
+			}
+
+			return runutil.Repeat(5*time.Minute, ctx.Done(), func() error {
+
+				if err := sy.SyncMetas(ctx); err != nil {
+					return errors.Wrapf(err, "could not sync metas")
+				}
+
+				metas := sy.Metas()
+				groups, err := grouper.Groups(metas)
+				if err != nil {
+					return errors.Wrapf(err, "could not group original metadata")
+				}
+
+				ps.CompactProgressMetrics.NumberOfCompactionRuns.Reset()
+				ps.CompactProgressMetrics.NumberOfCompactionBlocks.Reset()
+
+				if err = ps.ProgressCalculate(ctx, groups); err != nil {
+					return errors.Wrapf(err, "could not calculate compaction progress")
+				}
+
+				if !conf.disableDownsampling {
+					groups, err = grouper.Groups(metas)
+					if err != nil {
+						return errors.Wrapf(err, "could not group original metadata into downsample groups")
+					}
+					ds.DownsampleProgressMetrics.NumberOfBlocksDownsampled.Reset()
```

Comment:
ditto. This should be the implementation detail of the `ProgresssCalculate` method and let's put reset inside that method.